apps/dashboard/src/components/workflow-editor/steps/conditions/edit-step-conditions-form.tsx

```diff
@@ -0,0 +1,155 @@
+import { useMemo } from 'react';
+import { RiInputField, RiQuestionLine } from 'react-icons/ri';
+import { Link, useBlocker } from 'react-router-dom';
+import { formatQuery, RQBJsonLogic, RuleGroupType, RuleType } from 'react-querybuilder';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { prepareRuleGroup } from 'react-querybuilder';
+import { parseJsonLogic } from 'react-querybuilder/parseJsonLogic';
+import type { StepUpdateDto } from '@novu/shared';
+
+import { Button } from '@/components/primitives/button';
+import { Panel, PanelContent, PanelHeader } from '@/components/primitives/panel';
+import { ConditionsEditor } from '@/components/conditions-editor/conditions-editor';
+import { Form, FormField } from '@/components/primitives/form/form';
+import { parseStepVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { updateStepInWorkflow } from '@/components/workflow-editor/step-utils';
+import { useWorkflow } from '@/components/workflow-editor/workflow-provider';
+import { UnsavedChangesAlertDialog } from '@/components/unsaved-changes-alert-dialog';
+import { useBeforeUnload } from '@/hooks/use-before-unload';
+
+const rulesSchema: z.ZodType<RuleType | RuleGroupType> = z.union([
```

Comment:
the schema validation logic for the single rule which could be a rule object or a group with rules